### PR TITLE
InnerHTMLParserFastpath to release 10 production

### PR DESCRIPTION
**Pull request by atuchin-m**
For https://github.com/brave/brave-variations/issues/652
PR to main: https://github.com/brave/brave-variations/pull/668

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
